src/runtime_src/core/edge/user/shim.h

```diff
@@ -232,8 +234,102 @@ class shim {
     {
       m_shim->xclExecBuf(cmd->get_xcl_handle());
     }
+    std::shared_ptr<xrt_core::graph_handle>
+    open_graph_handle(const char* name, xrt::graph::access_mode am) override
+    {
+      return std::make_shared<graph_object>(m_shim, m_uuid, name, am);
+    }
   }; // class hwcontext
 
+  // Shim handle for graph object
+  class graph_object : public xrt_core::graph_handle
+  {
+    shim* m_shim;
+    xclGraphHandle m_xclGraphHandle;
+  
+  public:
+    graph_object(shim* shim, const xrt::uuid& uuid , const char* name, xrt::graph::access_mode am)
+      : m_shim(shim)
+    {
+      m_xclGraphHandle = xclGraphOpen(m_shim, uuid.get(), name, am);
+    }
+
+    ~graph_object()
+    {
+      (void)xclGraphClose(m_xclGraphHandle);
+    }
+
+    void
+    reset_graph() override
+    {
+      if (auto ret = xclGraphReset(m_xclGraphHandle))
+      {
+         throw xrt_core::system_error(ret, "fail to reset graph");
+      }
+    }
+
+    uint64_t
+    get_timestamp() override
+    {
+      return xclGraphTimeStamp(m_xclGraphHandle);
+    }
+
+    void
+    run_graph(int iterations) override
+    {
+      if (auto ret = xclGraphRun(m_xclGraphHandle, iterations))
+      throw xrt_core::system_error(ret, "fail to run graph");
+    }
+
+    int
+    wait_graph_done(int timeout) override
+    {
+      return xclGraphWaitDone(m_xclGraphHandle, timeout);
+    }
+
+    void
+    wait_graph(uint64_t cycle) override
+    {
+      if (auto ret = xclGraphWait(m_xclGraphHandle, cycle))
+      throw xrt_core::system_error(ret, "fail to wait graph");
```

Comment:
```suggestion
      if (auto ret = xclGraphWait(m_xclGraphHandle, cycle))
        throw xrt_core::system_error(ret, "fail to wait graph");
```
Please fix indentation for all such cases.